tensorflow_datasets/translate/en_tam_parallel_text.py

```diff
@@ -0,0 +1,253 @@
+    'http://opus.nlpl.eu/download.php?f=GNOME/v1/moses/en_GB-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=GNOME/v1/moses/en_US-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=KDE4/v2/moses/en-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=KDE4/v2/moses/en_GB-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Tatoeba/v20190709/moses/en-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en-ta_LK.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_GB-ta_LK.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_AU-ta_LK.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_CA-ta_LK.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_US-ta_LK.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_GB-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_AU-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_CA-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_NZ-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=Ubuntu/v14.10/moses/en_US-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=OpenSubtitles/v2018/moses/en-ta.txt.zip',
+    'http://opus.nlpl.eu/download.php?f=OpenSubtitles/v2016/moses/en-ta.txt.zip']
+
+_VALID_LANGUAGE_PAIRS = [
+    ("en", "ta"),
+    ("en_AU", "ta"),
+    ("en_CA", "ta"),
+    ("en_GB", "ta"),
+    ("en_NZ", "ta"),
+    ("en_US", "ta"),
+    ("en", "ta_LK"),
+    ("en_AU", "ta_LK"),
+    ("en_CA", "ta_LK"),
+    ("en_GB", "ta_LK"),
+    ('en_US', 'ta_LK')
+]
+
+
+class EnTamParallelTextConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for English & Tamil parallel text data."""
+
+  @api_utils.disallow_positional_args
+  def __init__(self, download_link=None, **kwargs):
+    """
+    Create source and target filenames from the link which
+	can be consumed by the download & extract function
+    Args:
+      download_link: links of the files to be downloaded
+      **kwargs: keyword arguments forwarded to super.
+      
+    """
+    self.language_pair = ("en", "ta")
+    if 'opus' in download_link:
+      language_pair = download_link.split('/')[-1].split('.')[0].split('-')
+      language_pair = tuple(language_pair)
+      typ_name = download_link.split('?f=')[1].split('/moses')[0].replace('/', '_')
+      name = "%s_to_%s" % (language_pair[0], language_pair[1])
+      name = typ_name+'_'+name
+      description = ("Translation dataset from %s to %s in plain text.") % (
+          language_pair[0], language_pair[1])
+      self.citation = _CITATION['opus']
+      self.descrp = DESCRIPTION['opus']
+      # Validate language pair.
+      assert language_pair in _VALID_LANGUAGE_PAIRS, (
+          "Config language pair (%s, "
+          "%s) not supported") % language_pair
+      if 'GNOME' in download_link:
```

Comment:
You used too many if else please don't repeat yourself and use loop.

Reply:
Modified the code as suggested. Please review :)